oauth_provider/models/oauth_provider_token.py

```diff
@@ -0,0 +1,98 @@
+# Copyright 2016 SYLEAM
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from odoo import models, api, fields, exceptions, _
+
+
+class OAuthProviderToken(models.Model):
+    _name = 'oauth.provider.token'
+    _description = 'OAuth Provider Token'
+    _rec_name = 'token'
+
+    token = fields.Char(required=True, help='The token itself.')
+    token_type = fields.Selection(
+        selection=[('Bearer', 'Bearer')], required=True, default='Bearer',
+        help='Type of token stored. Currently, only the bearer token type is '
+        'available.')
+    refresh_token = fields.Char(
+        help='The refresh token, if applicable.')
+    client_id = fields.Many2one(
+        comodel_name='oauth.provider.client', string='Client', required=True,
+        help='Client associated to this token.')
+    user_id = fields.Many2one(
+        comodel_name='res.users', string='User', required=True,
+        help='User associated to this token.')
+    scope_ids = fields.Many2many(
+        comodel_name='oauth.provider.scope', string='Scopes',
+        help='Scopes allowed by this token.')
+    expires_at = fields.Datetime(
+        required=True, help='Expiration time of the token.')
+    active = fields.Boolean(
+        compute='_compute_active', search='_search_active',
+        help='A token is active only if it has not yet expired.')
+
+    _sql_constraints = [
+        ('token_unique', 'UNIQUE (token, client_id)',
+         'The token must be unique per client !'),
+        ('refresh_token_unique', 'UNIQUE (refresh_token, client_id)',
+         'The refresh token must be unique per client !'),
+    ]
+
+    @api.multi
+    def _compute_active(self):
+        for token in self:
+            token.active = fields.Datetime.now() < token.expires_at
```

Comment:
Shouldn't this method be prefixed by `@api.depends("expires_at")` as the method use this field ?

Reply:
we need the dependency when we store the field, but we don't